devito/compiler.py

```diff
@@ -448,8 +448,7 @@ def __init__(self, *args, **kwargs):
         self.cflags.remove('-O3')
         self.cflags.remove('-Wall')
         self.cflags += ['-fast', '-acc']
-        # self.cflags += ['-Minfo=accel'] # Print informational messages for compliation
-        # self.cflags += ['-ta=tesla']    # Compile for a target GPU architecture
+        self.cflags += ['-ta=tesla,host']    # Compile for a target GPU architecture
```

Comment:
Isn't it purely tesla specific? If yes maybe should stay commented.

Reply:
The targets above are instructing the compiler for gpu code and single threaded cpu code.  This is also the default for the PGI openacc compiler.

Reply:
yeah... but why 'tesla' specifically? that looks a bit odd

how could we sniff the underlying gpu architecture?

Reply:
I do not have code that checks for another architecture, so the user would need to change.  Maybe you could use lspci or lsmod for linux.

Reply:
doesn't the PGI compiler have an option to return info about the underlying GPU architecture? for example, gcc has one, but only for CPUs

> Maybe you could use lspci or lsmod for linux.

so we either enhance `get_platform`, see [here](https://github.com/devitocodes/devito/blob/master/devito/archinfo.py#L141), or we'll have to take this line back:

```
self.cflags += ['-ta=tesla,host']
```

(commented out would be OK)

I'll see if I can get one of our summer students to work on this

Reply:
comment and commented out as this is the default target for PGI.  The line of code is left in  place commented out as others may want to adjust the target in the future.